### PR TITLE
Modify document of clickhouse backend to work with spark 3.2

### PR DESCRIPTION
**Pull request by zhanglistar**
## What changes were proposed in this pull request?

Just doc of how to build and run gluten with clickhouse backend in version of spark 3.2.

## How was this patch tested?

Just doc, tested in local env.

### CODE REVIEW

File: docs/ClickHouse.md [hunk 1]
```diff
@@ -121,20 +121,19 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
...
-rm -f jars/protobuf-java-2.5.0.jar ./
-rm -f jars/flatbuffers-java-1.9.0.jar ./
-#download protobuf-java-3.13.0.jar and flatbuffers-java-1.12.0.jar from maven
-cp protobuf-java-3.13.0.jar jars/
-cp flatbuffers-java-1.12.0.jar jars/
+#### Deploy Spark 3.2.2
+```
+tar zxf spark-3.2.2-bin-hadoop2.7.tgz
+cd spark-3.2.2-bin-hadoop2.7
+rm -f jars/arrow-*
```
Comment (zzcclp) [inline docs/ClickHouse.md:128]: don't need remove arrow-*.jar.

File: docs/ClickHouse.md [hunk 2]
```diff
@@ -121,20 +121,19 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
...
+rm -f jars/protobuf-java-2.5.0.jar
+rm -f jars/flatbuffers-java-1.9.0.jar
```
Comment (zzcclp) [inline docs/ClickHouse.md:130]: don't need remove flatbuffers-java-1.9.0.jar and update.

File: docs/ClickHouse.md [hunk 3]
```diff
@@ -121,20 +121,19 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
...
+#download protobuf-java-3.13.0.jar, flatbuffers-java-1.12.0.jar and delta-core_2.12-1.2.1.jar
```
Comment (zzcclp) [inline docs/ClickHouse.md:131]: add ` and delta-storage-1.2.1.jar`

File: docs/ClickHouse.md [hunk 4]
```diff
@@ -121,20 +121,19 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
...
+wget https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.13.0/protobuf-java-3.13.0.jar -P ./jars -O protobuf-java-3.13.0.jar
+wget https://repo1.maven.org/maven2/com/google/flatbuffers/flatbuffers-java/1.12.0/flatbuffers-java-1.12.0.jar -P ./jars -O flatbuffers-java-1.12.0.jar
```
Comment (zzcclp) [inline docs/ClickHouse.md:133]: remove

File: docs/ClickHouse.md [hunk 5]
```diff
@@ -121,20 +121,19 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
...
+wget https://repo1.maven.org/maven2/io/delta/delta-core_2.12/1.2.1/delta-core_2.12-1.2.1.jar -P ./jars -O delta-core_2.12-1.2.1.jar
+wget https://repo1.maven.org/maven2/io/delta/delta-storage/1.2.1/delta-storage-1.2.1.jar -P ./jars -O delta-storage-1.2.1.jar
 cp gluten-jvm-XXXXX-jar-with-dependencies.jar jars/
```
Comment (zzcclp) [inline docs/ClickHouse.md:136]: rename to `gluten-XXXXX-jar-with-dependencies.jar`

**Comment by zzcclp** [comment]
BTW, there are some other parts of the document needed to be modified, could you help to modify too?
For example: [run-spark-thriftserver-on-local](https://github.com/oap-project/gluten/blob/main/docs/ClickHouse.md#run-spark-thriftserver-on-local)

**Comment by zhanglistar** [comment]
> 

@zzcclp All done, check again, thanks.

**Comment by zzcclp** [comment]
please modify `based on version **21.9**` above to `based on version **22.3**`

File: docs/ClickHouse.md [hunk 6]
```diff
@@ -114,27 +114,23 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
     git clone https://github.com/oap-project/gluten.git
     cd gluten/
     export MAVEN_OPTS="-Xmx8g -XX:ReservedCodeCacheSize=2g"
-    mvn clean install -Pbackends-clickhouse -Phadoop-2.7.4 -Pspark-3.1.1 -Dhadoop.version=2.8.5 -DskipTests -Dbuild_cpp=OFF -Dcpp_tests=OFF -Dbuild_arrow=ON -Dbuild_protobuf=ON -Dbuild_jemalloc=ON -Dcheckstyle.skip
+    mvn clean install -Pbackends-clickhouse -Phadoop-2.7.4 -Pspark-3.2.2 -Dhadoop.version=2.8.5 -DskipTests -Dbuild_cpp=OFF -Dcpp_tests=OFF -Dbuild_arrow=ON -Dbuild_protobuf=ON -Dbuild_jemalloc=ON -Dcheckstyle.skip
```
Comment (zzcclp) [inline docs/ClickHouse.md:117]: `mvn clean install -Pbackends-clickhouse -Phadoop-2.7.4 -Pspark-3.2 -Dhadoop.version=2.8.5 -DskipTests -Dcheckstyle.skip`, remove `-Dbuild_cpp=OFF -Dcpp_tests=OFF -Dbuild_arrow=ON -Dbuild_protobuf=ON -Dbuild_jemalloc=ON`

**Comment by zzcclp** [comment]
and in section `Data preparation`, it needs to download a specified clickhouse version:
`sudo apt install -y --allow-downgrades clickhouse-server=22.5.1.2079 clickhouse-client=22.5.1.2079 clickhouse-common-static=22.5.1.2079`
now the converting script don't support the latest clickhouse version.

File: docs/ClickHouse.md [hunk 7]
```diff
@@ -179,7 +175,7 @@ cd spark-3.1.1-bin-2.8.5
   --conf spark.plugins=io.glutenproject.GlutenPlugin \
   --conf spark.gluten.sql.columnar.columnartorow=false \
```
Comment (zzcclp) [inline docs/ClickHouse.md:176]: changs to
```
--conf spark.gluten.sql.columnar.columnartorow=true \
  --conf spark.gluten.sql.columnar.loadnative=true \
  --conf spark.gluten.sql.columnar.libpath=/path_to_clickhouse_library/libch.so \
  --conf spark.gluten.sql.columnar.iterator=true \
  --conf spark.gluten.sql.columnar.loadarrow=false \
  --conf spark.gluten.sql.columnar.backend.lib=ch \
  --conf spark.gluten.sql.columnar.hashagg.enablefinal=true \
  --conf spark.gluten.sql.enable.native.validation=false \
  --conf spark.io.compression.codec=snappy \
  --conf spark.gluten.sql.columnar.backend.ch.use.v2=false \
  --conf spark.gluten.sql.columnar.forceshuffledhashjoin=true \
```

and the same as below in section `Run Spark Thriftserver` of the `Deploy on cloud`

**Comment by zhanglistar** [comment]
@zzcclp Check again pls.

File: docs/ClickHouse.md [hunk 8]
```diff
@@ -114,27 +114,23 @@ The prerequisites are the same as the one above mentioned. Compile Gluten with C
     git clone https://github.com/oap-project/gluten.git
     cd gluten/
     export MAVEN_OPTS="-Xmx8g -XX:ReservedCodeCacheSize=2g"
-    mvn clean install -Pbackends-clickhouse -Phadoop-2.7.4 -Pspark-3.1.1 -Dhadoop.version=2.8.5 -DskipTests -Dbuild_cpp=OFF -Dcpp_tests=OFF -Dbuild_arrow=ON -Dbuild_protobuf=ON -Dbuild_jemalloc=ON -Dcheckstyle.skip
+    mvn clean install -Pbackends-clickhouse -Phadoop-2.7.4 -Pspark-3.2 -Dhadoop.version=2.8.5 -DskipTests -Dcheckstyle.skip
     ls -al jvm/target/gluten-jvm-XXXXX-jar-with-dependencies.jar
```
Comment (zzcclp) [inline docs/ClickHouse.md:118]: changes to `ls -al backends-clickhouse/target/gluten-XXXXX-jar-with-dependencies.jar`

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**